config/opal_setup_cc.m4

```diff
@@ -23,6 +23,115 @@ dnl
+
+    opal_prog_cc_c11_helper_CFLAGS_save=$CFLAGS
+    CFLAGS="$CFLAGS $1"
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 thread local storage], [opal_prog_cc_c11_helper__thread_local_available],
+                   [],[[static _Thread_local int  foo = 1;++foo;]])
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 atomic variables], [opal_prog_cc_c11_helper_atomic_var_available],
+                   [[#include <stdatomic.h>]], [[static atomic_long foo = 1;++foo;]])
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 _Atomic keyword], [opal_prog_cc_c11_helper__atomic_available],
+                   [[#include <stdatomic.h>]],[[static _Atomic long foo = 1;++foo;]])
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 _Generic keyword], [opal_prog_cc_c11_helper__generic_available],
+                   [[#define FOO(x) (_Generic (x, int: 1))]], [[static int x, y; y = FOO(x);]])
+
+    OPAL_CC_HELPER([if $CC $1 supports C11 _Static_assert], [opal_prog_cc_c11_helper__static_assert_available],
+                   [[#include <stdint.h>]],[[_Static_assert(sizeof(int64_t) == 8, "WTH");]])
+
+    AS_IF([test "x$opal_prog_cc_c11_helper_atomic_var_available" = "xyes"], [opal_proc_cc_c11_helper_tmp=1],
+          [opal_proc_cc_c11_helper_tmp=0])
+
+    AC_DEFINE_UNQUOTED([OPAL_C_HAVE_C11_ATOMIC], [$opal_proc_cc_c11_helper_tmp], [Whether C compiler supports C11 atomics])
+
+    AS_IF([test "x$opal_prog_cc_c11_helper__atomic_available" = "xyes"], [opal_proc_cc_c11_helper_tmp=1],
+          [opal_proc_cc_c11_helper_tmp=0])
+
+    AC_DEFINE_UNQUOTED([OPAL_C_HAVE__ATOMIC], [$opal_proc_cc_c11_helper_tmp], [Whether C compiler supports __Atomic])
+
+    dnl At this time Open MPI only requires thread local and the atomic convenience types
+    AS_IF([test "x$opal_prog_cc_c11_helper__thread_local_available" = "xyes" && test "x$opal_prog_cc_c11_helper_atomic_var_available" = "xyes"],
+          [$2], [$3])
+
+    CFLAGS=$opal_prog_cc_c11_helper_CFLAGS_save
+
+    OPAL_VAR_SCOPE_POP
+])
+
+AC_DEFUN([OPAL_PROG_CC_C11],[
+    OPAL_VAR_SCOPE_PUSH([opal_prog_cc_c11_flags])
+    if test -z "$opal_cv_c11_supported" ; then
+        opal_cv_c11_supported=no
+        opal_cv_c11_flag_required=yes
+
+        AC_MSG_CHECKING([if $CC requires a flag for C11])
+
+        AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#if __STDC_VERSION__ < 201112L
+#error "Need a flag for C11"
+#endif
+                                           ]],[])],
+                          [opal_cv_c11_flag_required=no])
+
+        AC_MSG_RESULT([$opal_cv_c11_flag_required])
+
+        if test "x$opal_cv_c11_flag_required" = "xno" ; then
+            AC_MSG_NOTICE([verifying $CC supports C11 without a flag])
+            OPAL_PROG_CC_C11_HELPER([], [], [opal_cv_c11_flag_required=yes])
+        fi
```

Comment:
Any reason not to move the 2 addition lines from 112/113 up into this block (where no flag is required), and then make the next block be an `else`?